app/models/player.rb

```diff
@@ -0,0 +1,25 @@
+class Player < ActiveRecord::Base
+  belongs_to :party
+
+  validates :name,                presence: true
+  validates :level,               presence: true
+  validates :armor_class,         presence: true
+  validates :passive_perception,  presence: true
+
+
+  def battle_setup(initiative, starting_health)
```

Comment:
These methods should ensure an exception is raised, typically I'll use 
    self.current_health = current_health + health_difference
    self.save! # this will ensure an exception is raised on validation failure

alternatively, use update_attributes! (but this is a little silly if you're not passing multiple attributes)